src/v/utils/dns_resolver.h

```diff
@@ -0,0 +1,23 @@
+
+#pragma once
+
+#include "seastarx.h"
+#include "utils/mutex.h"
+#include "utils/unresolved_address.h"
+
+#include <seastar/net/dns.hh>
+
+class dns_resolver {
```

Comment:
Since this is all static, can the api just be:

```cpp
ss::future<ss::socket_address> dns_resolve(unresolved_address)
```
and the impl:
```cpp
ss::future<ss::socket_address>
dns_resolver::resolve(unresolved_address address) {
    static thread_local ss::net::dns_resolver resolver;
    static thread_local mutex m;
    // lock
    auto units = co_await get_mutex().get_units();
    // resolve
    auto i_a = co_await resolver().resolve_name(address.host(), {});

    co_return ss::socket_address(i_a, address.port());
}
```

Reply:
good point, will change

Reply:
And maybe moved to wherever `unresolved_address` lives.